test/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs

```diff
@@ -506,5 +437,123 @@ public class NuGetUpdateCommandTests
+
+                var projectFile = Path.Combine(projectDirectory, "proj1.csproj");
+                var solutionFile = Path.Combine(solutionDirectory, "a.sln");
+
+                var testNuGetProjectContext = new TestNuGetProjectContext();
+                var msbuildDirectory = MsBuildUtility.GetMsbuildDirectory("14.0", null);
+                var projectSystem = new MSBuildProjectSystem(msbuildDirectory, projectFile, testNuGetProjectContext);
+                var msBuildProject = new MSBuildNuGetProject(projectSystem, packagesDirectory, projectDirectory);
+                using (var stream = File.OpenRead(a1Package))
+                {
+                    var downloadResult = new DownloadResourceResult(stream);
+                    await msBuildProject.InstallPackageAsync(
+                        a1,
+                        downloadResult,
+                        testNuGetProjectContext,
+                        CancellationToken.None);
+                }
+
+                // Since, content files do not get added by MSBuildProjectSystem used by nuget.exe only,
+                // the csproj file will not have the entries for content files.
+                // Note that MSBuildProjectSystem used by nuget.exe only, can add or remove references from csproj.
+                // Replace proj1.csproj with content files on it, like Install-Package from VS would have.
+                File.Delete(projectFile);
+                File.WriteAllText(projectFile,
+                    Util.CreateProjFileContent(
+                        "proj1",
+                        "v4.5",
+                        references: null,
+                        contentFiles: new[] { "test1.txt" }));
+
+                var args = new[]
+                {
+                    "update",
+                    solutionFile,
+                    "-Source",
+                    packagesSourceDirectory
+                };
+
+                var test1textPath = Path.Combine(projectDirectory, "test1.txt");
+                var test2textPath = Path.Combine(projectDirectory, "test2.txt");
+                var packagesConfigPath = Path.Combine(projectDirectory, "packages.config");
+
+                Assert.True(File.Exists(test1textPath), "Content file test1.txt should exist but does not.");
+                Assert.False(File.Exists(test2textPath), "Content file test2.txt should not exist but does.");
+                using (var packagesConfigStream = File.OpenRead(packagesConfigPath))
+                {
+                    var packagesConfigReader = new PackagesConfigReader(packagesConfigStream);
+                    var packages = packagesConfigReader.GetPackages().ToList();
+                    Assert.Equal(1, packages.Count);
+                    Assert.Equal(a1, packages[0].PackageIdentity);
+                }
+
+                // Act
+                var r = CommandRunner.Run(
+                    Util.GetNuGetExePath(),
```

Comment:
Its not clear sometimes what is test code and what isn't. In particular the name Util is deceptive.